docs/source/notes/autograd.rst

```diff
@@ -890,11 +900,13 @@ The order in which the different hooks are fired
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 The order in which things happen are:
-1. hooks registered to Tensor are executed
-2. pre-hook registered to Node are executed (if Node is executed).
-3. The ``.grad`` field is updated for Tensors that retain_grad
-4. Node is executed (subject to rules above)
-5. post-hook registered to Node are executed (if Node is executed)
+
+#. hooks registered to Tensor are executed
+#. pre-hooks registered to Node are executed (if Node is executed).
+#. The ``.grad`` field is updated for Tensors that retain_grad
+#. Node is executed (subject to rules above)
+#. If the Node is AccumulateGrad, post accumulate grad hooks registered to Tensor are executed (if Node is executed)
```

Comment:
It's tricky to avoid mention of AccumulateGrad Node if we try to explain things this way. I'd be okay with making it a thing in the docs. Though @albanD you mentioned in the other PR that we may want to avoid mention of AccumulateGrad, curious is there's a different preferred way of explaining things like - 
```suggestion
#. for leaf Tensors that have ``.grad`` accumulated, post-accumulate-grad hooks are executed
```

Reply:
I think in the Note that talks about Nodes and how autograd works it is fine.
In the generally user-facing doc I would try to avoid it by just saying "when the .grad field is updated by autograd" as you suggested above